### PR TITLE
DO NOT MERGE: Just for teaching team : Render Draft

### PR DESCRIPTION
**Pull request by samcake**
Introducing the Scene class acting as a Database for the Render::Items. These are the representation side of the the various entities, avatars, or anything that needs to be displayed
 
any change to the scene is passed through the Scene""PendingCHanges which are all flushed and added to the scene database once per frame

THe rendering itself will be performed by a task working on the Scene

And now we have an Engine class yeah !

Now we have a lot to do to make it work but this is the initial design draft

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 26c7d63ee33c0c5d30a5c94ca3d843518656f986 available at: https://gist.github.com/1462ea7aa2206ad4cc3f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4873.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4873.dmg

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 26c7d63ee33c0c5d30a5c94ca3d843518656f986 available at: https://gist.github.com/1462ea7aa2206ad4cc3f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4873.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4873.dmg

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 26c7d63ee33c0c5d30a5c94ca3d843518656f986 available at: https://gist.github.com/1462ea7aa2206ad4cc3f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4873.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4873.dmg

**Comment by Atlante45** [comment]
@murillodigital this is sample code to show the team the API.
Not meant to be merged in that state.

File: libraries/render/src/render/Scene.h [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+        ~PayloadInterface() {}
+    protected:
+    };
+
+    template <class T> class Payload : public PayloadInterface {
+    public:
+        virtual const State&& getState() const { return getState<T>(*this); }
+        virtual const Bound&& getBound() const { return getBound<T>(*this); }
+        virtual void render(Context& context) { render<T>(this*, context); }
```
Comment (Atlante45) [inline libraries/render/src/render/Scene.h]: I'm curious, is this a typo or an obscure feature of C++?
Can't find anything online about `this*`.

Comment (samcake) [inline libraries/render/src/render/Scene.h]: It s a typo:)

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3893dff94f077728f62f0cf24253c18b70a19b53 available at: https://gist.github.com/1462ea7aa2206ad4cc3f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4873.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4873.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1a59f4a97ca5acc9c7a17a065efb507c4dd93e99 available at: https://gist.github.com/1462ea7aa2206ad4cc3f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4873.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4873.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 22c99d02b9f7ea1b23c7592e6ad15040c5638452 available at: https://gist.github.com/1462ea7aa2206ad4cc3f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4873.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4873.dmg

**Comment by ZappoMan** [comment]
Closing this because it was on the wrong branch. I've opened a new PR on correct branch...

**Pull request closed without merging**